projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/jetpack-contact-form/ui-handler/with-ui-handler-data-provider.tsx

```diff
@@ -68,6 +77,25 @@ const withUiHandlerDataProvider = createHigherOrderComponent( BlockListBlock =>
 			[ inputValue, isVisible, popoverProps, show, hide, toggle ]
 		);
 
+		const setContent = useCallback(
+			( newContent: string ) => {
+				// Remove the Jetpack Form block from the content.
+				const processedContent = newContent.replace(
+					/<!-- (\/)*wp:jetpack\/(contact-)*form ({.*} )*(\/)*-->/g,
+					''
+				);
+				const newContentBlocks = parse( processedContent );
+
+				// Check if the generated blocks are valid.
+				const validBlocks = newContentBlocks.filter( block => {
+					return block.isValid && block.name !== 'core/freeform';
+				} );
+				// Only update the valid blocks
+				replaceInnerBlocks( clientId, validBlocks );
+			},
+			[ clientId, replaceInnerBlocks ]
+		);
+
```

Comment:
```suggestion
		useAiContext( {
			onDone: setContent,
			onSuggestion: setContent,
			askQuestionOptions: { postId },
		} );

```

because of [this](https://github.com/Automattic/jetpack/pull/32209/files#r1281439708).